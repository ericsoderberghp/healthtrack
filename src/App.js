import { Grommet } from 'grommet';
import React, { useEffect, useState } from 'react';
import { Loading } from './components';
import Onboard from './Onboard';
import SignIn from './SignIn';
import Home from './Home';
import Categories from './Categories';
import CategoryAdd from './CategoryAdd';
import CategoryEdit from './CategoryEdit';
import Data from './Data';
import DataAdd from './DataAdd';
import DataEdit from './DataEdit';
import Correlate from './Correlate';
import NoteAdd from './NoteAdd';
import NoteEdit from './NoteEdit';
import Notes from './Notes';
import Router, { Route, Routes } from './Router';
import TrackContext from './TrackContext';
import theme from './theme';
import { useTrack } from './track';

const App = () => {
  const [themeMode, setThemeMode] = useState();
  // align the themeMode with the browser/OS setting
  useEffect(() => {
    if (window.matchMedia) {
      setThemeMode(
        window.matchMedia('(prefers-color-scheme: dark)').matches
          ? 'dark'
          : 'light',
      );
    }
  }, []);

  const [track, setTrack] = useTrack();

  const [nextPath, setNextPath] = useState();
  useEffect(() => {
    if (!nextPath && track !== undefined) {
      // initial load has finished, decide where to go
      if (track) setNextPath(window.location.pathname || '/');
      else if (track === false) setNextPath('/onboard');
    } else if (nextPath !== '/onboard' && track === false) {
      setNextPath('/onboard');
    } else if (nextPath === '/onboard' && track) {
      setNextPath('/');
    }
  }, [nextPath, track]);

  if (!nextPath) return <Loading />;

  return (
    <Router path={nextPath}>
      <Grommet
        theme={theme}
        themeMode={themeMode}
        style={{ minHeight: '100%' }}
      >
        <TrackContext.Provider value={[track, setTrack]}>
          <Routes redirect="/">
            <Route path="/onboard" Component={Onboard} />
            <Route path="/sign-in" Component={SignIn} />
            <Route path="/" Component={Home} />
            <Route path="/categories" Component={Categories} />
            <Route path="/categories/add" Component={CategoryAdd} />
            <Route path="/categories/:id" Component={CategoryEdit} />
            <Route path="/data" Component={Data} />
            <Route path="/data/add" Component={DataAdd} />
            <Route path="/data/:id" Component={DataEdit} />
            <Route path="/correlate" Component={Correlate} />
            <Route path="/notes" Component={Notes} />
            <Route path="/notes/add" Component={NoteAdd} />
            <Route path="/notes/:id" Component={NoteEdit} />
          </Routes>
        </TrackContext.Provider>
      </Grommet>
    </Router>
  );
};

export default App;
