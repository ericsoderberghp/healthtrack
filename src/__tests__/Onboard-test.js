import React from 'react';
import { render, cleanup } from '@testing-library/react';
import App from '../App';

delete window.location;
window.location = new URL('http://localhost/onboard');

afterEach(cleanup);

it('renders onboard', async () => {
  process.env.NODE_ENV = 'test';
  const { container, findByText } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
  await findByText('Hi!');
  expect(container.firstChild).toMatchSnapshot();
});
