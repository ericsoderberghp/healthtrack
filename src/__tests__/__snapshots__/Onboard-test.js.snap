// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders onboard 1`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fZoPPO"
  style="min-height: 100%;"
>
  <div
    class="StyledBox-sc-13pk1d4-0 EZpdJ"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fJfQVO"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 bozgir"
      >
        Hi!
      </h1>
      <p
        class="StyledParagraph-tbetod-0 gfTpGC"
      >
        Are you ready to make decisions about your health based on more than just guesswork?
      </p>
      <form>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Your name
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="name"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Your email
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Choose a password
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 cnjoFA"
        >
          <button
            class="StyledButtonKind-sc-1vhfpnt-0 jiUGCi"
            kind="primary"
            title="get started"
            type="submit"
          >
            Get Started
          </button>
        </div>
      </form>
      <div
        class="StyledBox-sc-13pk1d4-0 jDyrvX"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          type="button"
        >
          Sign In
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders onboard 2`] = `
<div
  class="StyledGrommet-sc-19lkkz7-0 fZoPPO"
  style="min-height: 100%;"
>
  <div
    class="StyledBox-sc-13pk1d4-0 EZpdJ"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 fJfQVO"
    >
      <h1
        class="StyledHeading-sc-1rdh4aw-0 bozgir"
      >
        Hi!
      </h1>
      <p
        class="StyledParagraph-tbetod-0 gfTpGC"
      >
        Are you ready to make decisions about your health based on more than just guesswork?
      </p>
      <form>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Your name
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="name"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Your email
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="email"
                type="email"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 bDztYr FormField__FormFieldBox-m9hood-0 eenrNp"
        >
          <label
            class="StyledText-sc-1sadyjn-0 enbIRN"
          >
            Choose a password
          </label>
          <div
            class="StyledBox-sc-13pk1d4-0 kpwHpK FormField__FormFieldContentBox-m9hood-1 ckZStK"
          >
            <div
              class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
            >
              <input
                autocomplete="off"
                class="StyledTextInput-sc-1x30a0s-0 jsQcHW"
                name="password"
                type="password"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 cnjoFA"
        >
          <button
            class="StyledButtonKind-sc-1vhfpnt-0 jiUGCi"
            kind="primary"
            title="get started"
            type="submit"
          >
            Get Started
          </button>
        </div>
      </form>
      <div
        class="StyledBox-sc-13pk1d4-0 jDyrvX"
      >
        <button
          class="StyledButtonKind-sc-1vhfpnt-0 Gjrpb"
          kind="default"
          type="button"
        >
          Sign In
        </button>
      </div>
    </div>
  </div>
</div>
`;
